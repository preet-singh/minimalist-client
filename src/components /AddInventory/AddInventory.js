import React, { Component } from "react";
import InventoryContext from '../../InventoryContext';
import config from "../../config";
import PropTypes from 'prop-types';

export default class AddInventory extends Component {
  state = {
    name: ''
  }

  static contextType = InventoryContext;

  handleAddInventory = (e) => {
    e.preventDefault();

    const newInventory = {
      inventory_name: this.state.name
    };

    fetch(`${config.API_ENDPOINT}/inventory`, {
      method: "POST",
      headers: {
        "content-type": "application/json"
      },
      body: JSON.stringify(newInventory)
    })
      .then(res => {
        if (!res.ok) return res.json().then(e => Promise.reject(e));
      })
      .then(() => {
        this.context.addInventory(newInventory);
        this.props.history.push('/');
      })
      .catch(error => {
        console.error({ error });
      });
  };

  getInventoryName = (e) => {
    this.setState({name: e.target.value});
  }

  validateInventoryName = () => {
    let inventoryName = this.state.name;

    if (!inventoryName) {
        return 'Inventory name is required'
    } else {
        return null
    }
}

  render() {
    return (
      <form className='addItemOrInventory' onSubmit={e => this.handleAddInventory(e)}>
        <div>
          <label htmlFor="inventoryName">New Inventory Name: </label>
          <input type="text" id="inventoryName" value={this.state.name} onChange={ this.getInventoryName } />
          {this.validateInventoryName && <p className='validationElement'>{this.validateInventoryName()}</p>}
          <button disabled={this.validateInventoryName()} type="submit">Submit</button>
        </div>
      </form>
    );
  }
}

AddInventory.propTypes = {
  history: PropTypes.object.isRequired
}