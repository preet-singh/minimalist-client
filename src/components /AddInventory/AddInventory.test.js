import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import AddInventory from './AddInventory';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Add inventory testing', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <BrowserRouter>
                <AddInventory />
            </BrowserRouter>,
        div)
    ReactDOM.unmountComponentAtNode(div)
    });

    it('renders the user inventory', () => {
        const wrapper = shallow(<AddInventory  />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});