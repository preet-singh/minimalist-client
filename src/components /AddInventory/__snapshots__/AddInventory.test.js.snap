// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add inventory testing renders the user inventory 1`] = `
<form
  className="addItemOrInventory"
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="inventoryName"
    >
      New Inventory Name: 
    </label>
    <input
      id="inventoryName"
      onChange={[Function]}
      type="text"
      value=""
    />
    <p
      className="validationElement"
    >
      Inventory name is required
    </p>
    <button
      disabled="Inventory name is required"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
